saito-core/src/core/consensus/transaction.rs

```diff
@@ -1055,60 +1154,409 @@ impl Transaction {
+                        "Create-bound transaction: slip2 nor normal slip {:?}",
+                        self.to.len()
+                    );
+                    return false;
+                }
+
+                //
+                // slip3 = zero amount
+                //
+                if self.to[2].amount != 0 {
+                    error!(
+                        "Create-bound transaction: output 2 (tracking slip) amount is not zero (found {}).",
+                        self.to[2].amount
+                    );
+                    return false;
+                }
 
-        if transaction_type == TransactionType::Bound {
-            // TODO : check if bound slips have matching normal slips
+                //
+                // any additional slips are not BoundSlips
+                //
+                // outputs[3..] = Normal
+                //
+                for slip in self.from.iter().skip(3) {
+                    if slip.slip_type != SlipType::Normal {
+                        error!(
+                            "Bound Transaction: created tx has unexpected non-normal slip (found {:?}).",
+                            slip.slip_type
+                        );
+                        return false;
+                    }
+                }
 
-            // first input slip should be non-zero
-            if let Some(slip) = self.from.first() {
-                if slip.amount == 0 {
-                    // first slip should have a value to make sure the NFT creator cannot mint multiple transactions with the same hash
+                //
+                // This section ensures that the bound slip (output[2]) truly encodes
+                // the unique UTXO that was consumed to mint this NFT. We decode the 33‑byte
+                // public_key on output[2] to extract:
+                //
+                //  - rec_block_id   – the original block_id (bytes 0..8)
+                //  - rec_tx_ord     – the original transaction ordinal (bytes 8..16)
+                //  - rec_slip_id    – the original slip_index (byte 16)
+                //
+                // We then compare these directly against the values on the slip we burned
+                // (self.from[0]). If any differ, the NFT‑UUID was forged or tampered with.
+                //
+
+                // Extract the 33‑byte “UUID” from the third output slip
+                let uuid_pk = self.to[2].public_key;
+
+                // 1) Decode original block_id (8 bytes, big-endian)
+                let rec_block_id = u64::from_be_bytes(uuid_pk[0..8].try_into().unwrap());
+
+                // 2) Decode original transaction ordinal (next 8 bytes)
+                let rec_tx_ord = u64::from_be_bytes(uuid_pk[8..16].try_into().unwrap());
+
+                // 3) Decode original slip_index (1 byte)
+                let rec_slip_id = uuid_pk[16];
+
+                // The slip we actually consumed to mint this NFT
+                let original_input = &self.from[0];
```

Comment:
does this match for ATRed txs also? line : 426 in create_rebroadcast_bound_transaction ?